File: problems/6/solution.py
"""Problem 5: Sum Square Difference"""

n = 100

nums = [i for i in range(1, n+1)]
sq_nums = [i ** 2 for i in range(1, n+1)]

ans = sum(nums) ** 2 - sum(sq_nums)

print(f"The difference between the sum of the squares of the first {n} natural numbers and the square of the sum is {ans}.")

File: problems/6/solution.go
package main

import "fmt"

func sumOfSquares(n int) int {
  sum := 0
  for i := 1; i <= n; i++ {
    sum += i * i
  }
  return sum
}

func squareOfSum(n int) int {
  sum := 0
  for i := 1; i <= n; i++ {
    sum += i
  }
  return sum * sum
}

func main() {
  const n = 100
  difference := squareOfSum(n) - sumOfSquares(n)
  fmt.Printf("Difference: %d\n", difference)
}

